.github/workflows/workflow-vercel-deployment.yml

```diff
@@ -0,0 +1,35 @@
+name: Vercel Preview Deployment
+
+env:
+  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+  VERCEL_PROJECT_ID: ${{ secrets.VERCEL_PROJECT_ID }}
+
+on:
+    workflow_call:
+
+permissions:
+    checks: read
+    contents: read
+    pull-requests: write
+
+jobs:
+  Deploy-Preview-and-Comment:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Vercel CLI
+        run: npm install --global vercel@latest
+      - name: Pull Vercel Environment Information
+        run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
+      - name: Build Project Artifacts
+        run: vercel build --token=${{ secrets.VERCEL_TOKEN }}
+      - name: Deploy Project Artifacts to Vercel
+        run: vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }}
+
+      - run: sleep 10
```

Comment:
https://github.com/marketplace/actions/vercel-preview-url
this workflow's readme said
"Vercel needs a little time to build the preview, you can check the average build time in your deployments and add the seconds plus a little to a sleep action, to wait until the deployment is READY."
I think I got confused since he are not using the preview link but the direct link so I removed it